test/lib/misc-junk.test.js

```diff
@@ -0,0 +1,109 @@
+const test = require('tape')
+const { getHeads } = require('../../lib/misc-junk')
+
+test('getHeads: Small graph with one head', t => {
+  //    A   (root)
+  //    |
+  //    B
+
+  const inputMap = {
+    A: { B: 1 }
+  }
+
+  const expectedHeads = ['B']
+
+  t.deepEqual(getHeads(inputMap), expectedHeads, 'one head')
+
+  t.end()
+})
+
+test('getHeads: Root-only graph', t => {
+  //    A   (root)
+
+  const inputMap = {
+    A: {}
+  }
+
+  const expectedHeads = ['A']
+
+  t.deepEqual(getHeads(inputMap), expectedHeads, 'root-only')
+
+  t.end()
+})
+
+test('getHeads: Root and dangle', t => {
+  //    A   (root)
+  //     
+  //          ----?--- J? (a message we don't have)
+  //                   |
+  //                   K
+
+  const inputMap = {
+    A: {},
+    J: { K: 1 }
+  }
+
+  const expectedHeads = ['K']
```

Comment:
Ok had to add this one because it was fucking with my head. I think a naive (or maybe reasonable) algorithm would respond `['A', 'K']` since they look like heads at first glance. But I think `['K']` is technically correct since `A` is the root and so we know that `K` is a descendant of it (i.e. `A` is not a head).

Also hehe I think accidentally this test is passing, but I think that's for the same reason that `getHeads: Root-only graph` is failing: the root is ignored.

Reply:
Ok there's no way for getHeads to know that A is the root, plus if this map/head logic should work for sub(partial?) graphs then it also sort of makes sense to have both as heads.

So I think I think this line should be changed to `['A', 'K']`